File: Selenium/Activities/Activity20.py
"""Open a new browser to https://training-support.net/webelements/alerts
Get the title of the page and print it to the console.
Find the button to open a PROMPT alert and click it.
Switch the focus from the main window to the Alert box and get the text in it and print it.
Type "Awesome!" into the prompt.
Close the alert by clicking Ok.
Close the browser."""

#Import the web driver
from selenium import webdriver
from selenium.webdriver.common.by import By
from selenium.webdriver.support import expected_conditions as EC
from selenium.webdriver.support.wait import WebDriverWait

#Initialize the firefox driver
with webdriver.Firefox() as driver:
    
    #Declare the wait variable
    wait = WebDriverWait(driver, timeout=10)

    #Open the web page and print the title of the pae
    driver.get("https://training-support.net/webelements/alerts")
    print("Title of the web page: ", driver.title)

    #Find the prompt alert and click it
    driver.find_element(By.ID, "prompt").click()

    #Switch focus to alert
    promptAlert = wait.until(EC.alert_is_present())

    #Print the text in the prompt alert box
    print("Text in the prompt alert text box: ", promptAlert.text)

    #Enter Awesome! in the prompt alert box
    promptAlert.send_keys("Awesome!")

    #Close the alert by clicking ok
    promptAlert.accept()

    #Print the message
    print(driver.find_element(By.ID, "result").text)

File: Selenium/Activities/Activity20.java
/*
 Open a new browser to https://training-support.net/webelements/alerts
Get the title of the page and print it to the console.
Find the button to open a PROMPT alert and click it.
Switch the focus from the main window to the Alert box and get the text in it and print it.
Type "Awesome!" into the prompt.
Close the alert by clicking Ok.
Close the browser.
 */

package training;

import org.openqa.selenium.Alert;
import org.openqa.selenium.By;
import org.openqa.selenium.WebDriver;
import org.openqa.selenium.firefox.FirefoxDriver;

public class Activity20 {
	
	public static void main(String[] args) throws InterruptedException {
		
		//Create new instance for firefox driver
		WebDriver driver = new FirefoxDriver();
		
		//Open the web page and print the title of the page
		driver.get("https://training-support.net/webelements/alerts");
		System.out.println("Title of the web page: " + driver.getTitle());
		
		//Find the button to open a PROMPT alert and click it
		driver.findElement(By.id("prompt")).click();
		
		//Switch focus to the alert box
		Alert promptAlert = driver.switchTo().alert();
		System.out.println("Text in the alert box is: " + promptAlert.getText());
		
		//Type 'Awesome' in the alert prompt window
		promptAlert.sendKeys("Awesome!");
		Thread.sleep(5000);
		
		//Close the alert by clicking ok
		promptAlert.accept();
		System.out.println(driver.findElement(By.id("result")).getText());
		
		//Close the browser
		driver.quit();
	}
}
